plugins/permission-react/src/components/PermissionedRoute.tsx

```diff
@@ -28,13 +32,25 @@ import { Permission } from '@backstage/plugin-permission-common';
  */
 export const PermissionedRoute = (
   props: ComponentProps<typeof Route> & {
-    permission: Permission;
-    resourceRef?: string;
     errorComponent?: ReactElement | null;
-  },
+  } & (
+      | {
+          permission: Exclude<Permission, ResourcePermission>;
```

Comment:
The type `Exclude<Permission, ResourcePermission>` is used in a few places across these PRs, with the goal of making it easier to add new kinds of permission in the future. Using `BasicPermission` here would mean the type needed to change when a third type of permission was added. 

1. Do you buy this, or do you think it's likely gonna have to change anyway?
2. Do you think we should consider introducing a convenience type for this? I toyed with `type NonResourcePermission = Exclude<Permission, ResourcePermission>`...

Reply:
Hard to say before we have other types of permissions in mind. That's what pushes me toward keeping things more explicit, and maybe redundant, for now. I'll pull the changes locally and test a bit more, but don't want to block here since I think it's okay to ship this as is. I suggest throwing a comment around these types just to preserve that explanation though